src/api/c/graphics_common.cpp

```diff
@@ -178,15 +233,15 @@ void ForgeManager::setWindowChartGrid(const forge::Window* window,
 {
     ChartMapIter iter = mChartMap.find(window);
 
-        if(iter != mChartMap.end()) {
-
-    }
-
     if(iter != mChartMap.end()) {
         // ChartVec found. Clear it.
         // TODO: Should we clear this even if r = old_r and c = old_c?
-        for(int i = 0; i < (int)(iter->second).size(); i++)
-            if((iter->second)[i] != NULL) delete (iter->second)[i];
+        for(int i = 0; i < (int)(iter->second).size(); i++) {
```

Comment:
The only properties users can change are axes titles and limits, just setting them to default values does the trick. But of course, there is no convenience function to reset all values in one go, we have to call `setAxesLimits` and `setAxesTitles` individually.